File: src/main/java/com/greatdreams.kotlin.template/MainProgram.java
package com.greatdreams.kotlin.template;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class MainProgram {
    private static Logger logger = LoggerFactory.getLogger(MainProgram.class);

    public static void main(String[] args) {
        logger.debug("program begins to run...");
        logger.debug("program exits normally.");
    }
}

File: src/main/kotlin/com/greatdreams/kotlin/template/mainprogram.kt
package com.greatdreams.kotlin.template

import org.slf4j.LoggerFactory

/**
 * @author Greatdreams
 * Created by greatdreams on 1/9/17.
 * Use [kotlin.reflect.KClass.declaredMemberExtensionProperties] to enumerate
 */
object MainClass {
    val log = LoggerFactory.getLogger(javaClass)

    @JvmStatic fun main(args: Array<String>) {
        log.info("The main program begins to run.")
        log.info("The main program exits normally.")
    }
}